API: Cities

sgerrand opened the pull request:
💁 These changes:
* add support for the [Cities API](https://duffel.com/docs/api/v1/cities)
* alters `DuffelAPI::Resources::Airport.new.city` to return a `DuffelAPI::Resource::City`

⚠️ The test suite wouldn't run so to get that passing I've also made the following changes:
* updated `gc_ruboconfig`, `rubocop` gems
* fixed various offenses raised by RuboCop
* fixed a silently broken test case
* bumped the minimum Ruby version to 2.7 (this version is EOL)

sgerrand committed:
Implement Cities API

See: https://duffel.com/docs/api/v1/cities

sgerrand committed:
Support rendering embedded City resources

sgerrand committed:
`bundle update gc_ruboconfig rubocop`

sgerrand committed:
Fix RuboCop offense: Gemspec/DevelopmentDependencies

Gemspec/DevelopmentDependencies: Specify development dependencies in
Gemfile.

sgerrand committed:
Fix silently broken test case

Pull test case out of setup.

sgerrand committed:
CI: update Ruby test matrix

- Remove Ruby 2.6: rubocop >= 1.51.0 depends on Ruby >= 2.7.0
- Add modern, generally available Ruby versions: 3.2, 3.3

sgerrand committed:
`bundle exec appraisal update`

sgerrand commented:
👋 @ulissesalmeida, are you able to review these changes?

sgerrand committed:
Make examples time aware

Time moves on and the ages in these examples are no longer valid. These
changes replace static ages with calculated equivalents so that they can
be rerun at any time with confidence.

ulissesalmeida reviewed (approved):
@sgerrand Long time my friend!

That is tons of work!
Thank you. 

Looks good to me!

sgerrand commented:
Too long! Hope that you're doing well. 💝

sgerrand committed:
Manage GitHub Actions updates with Dependabot

See: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#configuration-options-for-the-dependabotyml-file

sgerrand committed:
Merge pull request #6 from sgerrand/dependabot-github-actions

Dependabot: Manage GitHub Actions updates

dependabot[bot] committed:
Bump actions/upload-artifact from 2 to 4

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 2 to 4.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v2...v4)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

sgerrand committed:
Correct RuboCop offense: Style/SuperArguments

Style/SuperArguments: Call super without arguments and parentheses when
the signature is identical.

sgerrand committed:
Correct RuboCop offense: Style/RedundantInterpolation

Style/RedundantInterpolation: Prefer to_s over string interpolation.

sgerrand committed:
Merge pull request #5 from sgerrand/time-aware-examples

Examples: time aware passenger ages

sgerrand committed:
Satisfy the RuboCop :cop:

- Style/SuperArguments: Call super without arguments and parentheses
  when the signature is identical.
- Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block
  body end. (https://rubystyle.guide#empty-lines-around-bodies)
